javaslang/src/main/java/javaslang/collection/List.java

```diff
@@ -85,9 +81,80 @@
  * @author Daniel Dietrich
  * @since 1.1.0
  */
-public interface List<T> extends Kind1<List<?>, T>, LinearSeq<T>, Stack<T> {
+public final class List<T> implements Stack<T>, LinearSeq<T>, Kind1<List<?>, T>, Serializable {
```

Comment:
This change started as a benchmark experiment, pushed it and said that it's "controversial" and that I will gladly split it into many, more cohesive commits, if needed; I pushed it to gather comments. Though I expected something else than "dude, you're not supposed to touch that, because that's how Haskell and Scala did it, so it's the way we should do it also - even if it might not be the best solution in Java".
Will split it into more commits.